src/ralph/virtual/admin.py

```diff
@@ -52,10 +55,28 @@ class VirtualServerTypeForm(RalphAdmin):
 
 
 class VirtualServerForm(RalphAdminForm):
+    HYPERVISOR_TYPE_ERR_MSG = 'Hypervisor must be one of \
+                                   DataCenterAsset or VirtualServer.'
+    HYPERVISOR_IS_SELF_ERR_MSG = 'Hypervisor and host must be different.'
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         if 'parent' in self.fields:
             self.fields['parent'].required = True
+            self.fields['parent'].validators.extend((
```

Comment:
sorry for the confusion - I mean `clean_parent` - if you name a method like that, [it should work out of the box](https://docs.djangoproject.com/en/2.0/ref/forms/validation/), without adding it "manually" to validators

Reply:
Done.